src/ImageMetadata.jl

```diff
@@ -66,87 +77,100 @@ AxisArrays.HasAxes(A::ImageMetaAxis) = AxisArrays.HasAxes{true}()
             ret
         end
         @inline function Base.getindex(img::$AType{T,N}, I::Vararg{Int,N}) where {T,N}
-            @boundscheck checkbounds(img.data, I...)
-            @inbounds ret = img.data[I...]
+            @boundscheck checkbounds(data(img), I...)
+            @inbounds ret = data(img)[I...]
             ret
         end
 
         @inline function Base.setindex!(img::$AType{T,1}, val, i::Int) where T
-            @boundscheck checkbounds(img.data, i)
-            @inbounds img.data[i] = val
+            @boundscheck checkbounds(data(img), i)
+            @inbounds data(img)[i] = val
             val
         end
         @inline function Base.setindex!(img::$AType, val, i::Int)
-            @boundscheck checkbounds(img.data, i)
-            @inbounds img.data[i] = val
+            @boundscheck checkbounds(data(img), i)
+            @inbounds data(img)[i] = val
             val
         end
         @inline function Base.setindex!(img::$AType{T,N}, val, I::Vararg{Int,N}) where {T,N}
-            @boundscheck checkbounds(img.data, I...)
-            @inbounds img.data[I...] = val
+            @boundscheck checkbounds(data(img), I...)
+            @inbounds data(img)[I...] = val
             val
         end
     end
 end
 
 @inline function Base.getindex(img::ImageMetaAxis, ax::Axis, I...)
-    result = img.data[ax, I...]
+    result = data(img)[ax, I...]
     maybe_wrap(img, result)
 end
 @inline function Base.getindex(img::ImageMetaAxis, i::Union{Integer,AbstractVector,Colon}, I...)
-    result = img.data[i, I...]
+    result = data(img)[i, I...]
     maybe_wrap(img, result)
 end
 maybe_wrap(img::ImageMeta{T}, result::T) where T = result
 maybe_wrap(img::ImageMeta{T}, result::AbstractArray{T}) where T = copyproperties(img, result)
 
 
 @inline function Base.setindex!(img::ImageMetaAxis, val, ax::Axis, I...)
-    setindex!(img.data, val, ax, I...)
+    setindex!(data(img), val, ax, I...)
 end
 @inline function Base.setindex!(img::ImageMetaAxis, val, i::Union{Integer,AbstractVector,Colon}, I...)
-    setindex!(img.data, val, i, I...)
+    setindex!(data(img), val, i, I...)
 end
 
-Base.view(img::ImageMeta, ax::Axis, I...) = shareproperties(img, view(img.data, ax, I...))
-Base.view(img::ImageMeta{T,N}, I::Vararg{ViewIndex,N}) where {T,N} = shareproperties(img, view(img.data, I...))
-Base.view(img::ImageMeta, i::ViewIndex) = shareproperties(img, view(img.data, i))
-Base.view(img::ImageMeta, I::Vararg{ViewIndex,N}) where {N} = shareproperties(img, view(img.data, I...))
+Base.view(img::ImageMeta, ax::Axis, I...) = shareproperties(img, view(data(img), ax, I...))
+Base.view(img::ImageMeta{T,N}, I::Vararg{ViewIndex,N}) where {T,N} = shareproperties(img, view(data(img), I...))
+Base.view(img::ImageMeta, i::ViewIndex) = shareproperties(img, view(data(img), i))
+Base.view(img::ImageMeta, I::Vararg{ViewIndex,N}) where {N} = shareproperties(img, view(data(img), I...))
+
+function Base.getproperty(img::ImageMeta, propname::Symbol)
```

Comment:
```suggestion
function getproperty(img::ImageMeta, propname::Symbol)
```

if we `import Base: getproperty`, this adds one more method to `Base.getproperty`; if we don't (for julia < v1.2), then `Base` doesn't get affected.

Reply:
Oh looks like I made a mistake here with `getproperty` and `hasproperty`. Since `isdefined(Base, :getproperty) == true` holds for all julia 1.x version, there's no need to do so.